spark/src/main/scala/org/apache/spark/sql/comet/CometScanExec.scala

```diff
@@ -141,8 +141,30 @@ case class CometScanExec(
     if (wrapped == null) Map.empty else wrapped.metadata
 
   override def verboseStringWithOperatorId(): String = {
-    getTagValue(QueryPlan.OP_ID_TAG).foreach(id => wrapped.setTagValue(QueryPlan.OP_ID_TAG, id))
-    wrapped.verboseStringWithOperatorId()
+    val metadataStr = metadata.toSeq.sorted
```

Comment:
Since the original issue was that `OP_ID_TAG` has been removed, is it not sufficient to just remove the offending line

Reply:
Just removing setting the `OP_ID_TAG` leads to a formatted explain of:

```
== Physical Plan ==
* ColumnarToRow (2)
+- CometScan parquet  (1)


(unknown) Scan parquet 
Output [20]: [_1#0, _2#1, _3#2, _4#3, _5#4L, _6#5, _7#6, _8#7, _9#8, _10#9, _11#10L, _12#11, _13#12, _14#13, _15#14, _16#15, _17#16, _18#17, _19#18, _20#19]
Batched: true
Location: InMemoryFileIndex [file:/Users/abinford/projects/arrow-datafusion-comet/spark/target/tmp/spark-c0b82b7c-3de1-431b-96bf-56cb37f3a463/test.parquet]
ReadSchema: struct<_1:boolean,_2:tinyint,_3:smallint,_4:int,_5:bigint,_6:float,_7:double,_8:string,_9:smallint,_10:int,_11:bigint,_12:decimal(20,0),_13:string,_14:binary,_15:decimal(5,2),_16:decimal(18,10),_17:decimal(38,37),_18:timestamp,_19:timestamp,_20:date>

(2) ColumnarToRow [codegen id : 1]
Input [20]: [_1#0, _2#1, _3#2, _4#3, _5#4L, _6#5, _7#6, _8#7, _9#8, _10#9, _11#10L, _12#11, _13#12, _14#13, _15#14, _16#15, _17#16, _18#17, _19#18, _20#19]
```

which is the whole reason it was added in the first place, to fix the unknown operator ID https://github.com/apache/datafusion-comet/commit/bc35fa5a415f4a137177ab4fc66bd0167f9f23d3

Reply:
Plus by what I could see, every other operator is prefixed with `Comet` in the formatted explain, so it's weird for Scan to be the one thing that doesn't actually match up to the physical plan

Reply:
I see, that makes sense.
On an unrelated note, wonder why `CometScanExec` extends `DataSourceScanExec` instead of `FileSourceScanExec` (if it had we would have got the verboseString for free)